src/os/user/iterate_unix_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build (aix || darwin || dragonfly || freebsd || (js && wasm) || (!android && linux) || netbsd || openbsd || solaris) && (!cgo || osusergo)
+// +build aix darwin dragonfly freebsd js,wasm !android,linux netbsd openbsd solaris
+// +build !cgo osusergo
+
+package user
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestIterateGroups(t *testing.T) {
+	const testGroupFile = `# See the opendirectoryd(8) man page for additional 
+# information about Open Directory.
+##
+nobody:*:-2:
+nogroup:*:-1:
+invalidgid:*:notanumber:root
++plussign:*:20:root
+    indented:*:7:
+# comment:*:4:found
+     # comment:*:4:found
+kmem:*:2:root
+`
+	// Ordered list of correctly parsed groups from testGroupFile
+	var groups = []*Group{
+		{
+			Gid:  "-2",
+			Name: "nobody",
+		},
+		{
+			Gid:  "-1",
+			Name: "nogroup",
+		},
+		{
+			Gid:  "7",
+			Name: "indented",
+		},
+		{
+			Gid:  "2",
+			Name: "kmem",
+		},
+	}
+
+	r := strings.NewReader(testGroupFile)
+
+	collectedGroups := make([]*Group, 0, 4)
```

Comment:
use `len(groups)` instead of `4`, since the tests may change, but this value may get out of sync.